### PR TITLE
Improvement/prevent duplicated anrs

### PR DESCRIPTION
**Pull request by konraddysput**
# Why

This diff 
1. Prevents situation when android java thread stuck and android watchdog wants to report more than one ANR
2. make sure network communication is always right - when coronerd doesn't return expected message with status code 200 (service unavailable or something) we're still able to convert JSON to data without unhandled exception,
3. Configurable ANR watchdog options

### CODE REVIEW

**Review by vlussenburg — APPROVED** [review]

File: Runtime/Native/Base/NativeClientBase.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@ internal abstract class NativeClientBase
         protected const string CrashType = "Crash";
         protected const string ErrorTypeAttribute = "error.type";
 
+        protected int AnrWatchdogTimeout = 5000;
```
Comment (rqbacktrace) [inline Runtime/Native/Base/NativeClientBase.cs:16]: Should we initialize this to `DefaultAnrWatchdogTimeout`?

File: Android/BacktraceANRWatchdog.java [hunk 2]
```diff
@@ -62,11 +62,11 @@
     /**
      * Initialize new instance of BacktraceANRWatchdog with default timeout
      */
-    public BacktraceANRWatchdog(String gameObjectName, String methodName) {
+    public BacktraceANRWatchdog(String gameObjectName, String methodName, int anrTimeout) {
         Log.d(LOG_TAG, "Initializing ANR watchdog");
         this.methodName = methodName;
         this.gameObjectName = gameObjectName;
-        this.timeout = DEFAULT_ANR_TIMEOUT;
```
Comment (rqbacktrace) [inline Android/BacktraceANRWatchdog.java:69]: Is this variable `DEFAULT_ANR_TIMEOUT` still used somewhere or can we delete it now? If it's still used, should we unify with `DefaultAnrWatchdogTimeout`?

File: Android/BacktraceANRWatchdog.java [hunk 3]
```diff
@@ -105,6 +108,11 @@ public void run() {
                         "is on and connected debugger");
                 continue;
             }
+            if(reported) {
```
Comment (rqbacktrace) [inline Android/BacktraceANRWatchdog.java]: Formatting: Should it be `if (reported)`?

**Review by rqbacktrace — APPROVED** [review]
Looks good to me!

**Review by perf2711 — APPROVED** [review]
LGTM!

File: Android/BacktraceANRWatchdog.java (hunk 2 above)
Comment (konraddysput) [reply Android/BacktraceANRWatchdog.java:69]: good point. thanks removed!

**Pull request merged**